File: tests/fortran/basics/hello_world.f90
! vybe-test: fortran/basics/hello_world
! origin: languages/fortran/tests/fortran/test_basics.rs

program hello
    if (trim("Hello, World!") /= "Hello, World!") then
    print *, "FAIL: want [Hello, World!] got [", "Hello, World!", "]"
    stop 1
end if
end program hello

File: tests/csharp/basics/hello_world.cs
// vybe-test: csharp/basics/hello_world
// origin: languages/csharp/tests/csharp/test_basics.rs

using static __Harness;

__P(("Hello, World!").ToString());
__Check("Hello, World!");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}
